sdk/textanalytics/Azure.AI.TextAnalytics/src/ServiceClients/LegacyServiceClient.cs

```diff
@@ -430,6 +434,8 @@ public override async Task<Response<PiiEntityCollection>> RecognizePiiEntitiesAs
         public override Response<PiiEntityCollection> RecognizePiiEntities(string document, string language = default, RecognizePiiEntitiesOptions options = default, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNullOrEmpty(document, nameof(document));
+            Validation.SupportsOperation(nameof(TextAnalyticsClient.RecognizePiiEntities), TextAnalyticsClientOptions.ServiceVersion.V3_1, ServiceVersion);
```

Comment:
Do you need the method name here?  Or could you use https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callermembernameattribute?view=net-6.0 instead?

Reply:
I actually considered that, but the exception message should include the method people called - not the implementation, which *might* change. Probably not, but I was being paranoid about it.